src/acceptance/app/recurring_schedule_policy_test.go

```diff
@@ -0,0 +1,156 @@
+	var (
+		appName              string
+		appGUID              string
+		instanceName         string
+		dayOfMonthOrWeek     string
+		location             *time.Location
+		startTime            time.Time
+		endTime              time.Time
+		policyByte           []byte
+	)
+
+	BeforeEach(func() {
+		instanceName = generator.PrefixedRandomName("autoscaler", "service")
+		createService := cf.Cf("create-service", cfg.ServiceName, cfg.ServicePlan, instanceName).Wait(cfg.DefaultTimeoutDuration())
+		Expect(createService).To(Exit(0), "failed creating service")
+
+		initialInstanceCount := 1
+		appName = generator.PrefixedRandomName("autoscaler", "nodeapp")
+		countStr := strconv.Itoa(initialInstanceCount)
+		createApp := cf.Cf("push", appName, "--no-start", "-i", countStr, "-b", cfg.NodejsBuildpackName, "-m", cfg.NodeMemoryLimit, "-p", config.NODE_APP, "-d", cfg.AppsDomain).Wait(cfg.DefaultTimeoutDuration())
+		Expect(createApp).To(Exit(0), "failed creating app")
+
+		guid := cf.Cf("app", appName, "--guid").Wait(cfg.DefaultTimeout)
+		Expect(guid).To(Exit(0))
+		appGUID = strings.TrimSpace(string(guid.Out.Contents()))
+
+		Expect(cf.Cf("start", appName).Wait(cfg.DefaultTimeout * 2)).To(Exit(0))
+		waitForNInstancesRunning(appGUID, initialInstanceCount, cfg.DefaultTimeoutDuration())
+	})
+
+	AfterEach(func() {
+		Expect(cf.Cf("delete", appName, "-f", "-r").Wait(cfg.CfPushTimeoutDuration())).To(Exit(0))
+
+		deleteService := cf.Cf("delete-service", instanceName, "-f").Wait(cfg.DefaultTimeoutDuration())
+		Expect(deleteService).To(Exit(0))
+	})
+
+	Context("when scale out by recurring schedule", func() {
+
+		JustBeforeEach(func() {
+			timeZone := "GMT"
+			location, _ = time.LoadLocation(timeZone)
+			startTime, endTime = getStartAndEndTime(location)
+
+			policyStr := setRecurringScheduleDateTime(policyByte, timeZone, startTime, endTime, dayOfMonthOrWeek)
+			bindService := cf.Cf("bind-service", appName, instanceName, "-c", policyStr).Wait(cfg.DefaultTimeoutDuration())
+			Expect(bindService).To(Exit(0), "failed binding service to app with a policy ")
+		})
+
+		AfterEach(func() {
+			unbindService := cf.Cf("unbind-service", appName, instanceName).Wait(cfg.DefaultTimeoutDuration())
+			Expect(unbindService).To(Exit(0), "failed unbinding service from app")
+		})
+
+		Context("with day of month", func() {
+			BeforeEach(func() {
+				var err error
+				dayOfMonthOrWeek = "dom"
+				policyByte, err = ioutil.ReadFile("../assets/file/policy/recurringschedule_dom.json")
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			It("should scale", func() {
+				totalTime := time.Duration(cfg.ReportInterval*2)*time.Second + 2*time.Minute
+				By("Start schedule")
+				waitForNInstancesRunning(appGUID, 3, totalTime)
+
+				By("Within schedule")
+				jobRunTime := endTime.Sub(time.Now().In(location))
+				Consistently(func() int {
+					return runningInstances(appGUID, jobRunTime)
+				}, jobRunTime, 15*time.Second).Should(BeNumerically(">=", 2))
```

Comment:
When scaling schedule kicks in, the instance count will be 3 because of "initial_min_instance_count": 3. 
Now the rule is 
scaling_rules": [
    {
      "metric_type": "memoryused",
      "stat_window_secs": 120,
      "breach_duration_secs": 120,
      "threshold": 1000,
      "operator": "<",
      "cool_down_secs": 60,
      "adjustment": "-1"
    }
  ]
So because of  threshold being very high the instance count will be changed to 2 within the schedule (as being checked Should(BeNumerically(">=", 2)).
And then it is set to instance_min_count the one which is outside the schedule.
Writing in detail just to understand if this how it is supposed to function or are we missing something.